instrumentation/spring/spring-boot-autoconfigure/src/main/javaSpring4/io/opentelemetry/instrumentation/spring/autoconfigure/internal/instrumentation/web/RestClientBeanPostProcessorSpring4.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.internal.instrumentation.web;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.incubator.config.internal.InstrumentationConfig;
+import io.opentelemetry.instrumentation.spring.autoconfigure.internal.properties.InstrumentationConfigUtil;
+import io.opentelemetry.instrumentation.spring.web.v3_1.SpringWebTelemetry;
+import io.opentelemetry.instrumentation.spring.web.v3_1.internal.WebTelemetryUtil;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.web.client.RestClient;
+
+final class RestClientBeanPostProcessorSpring4 implements BeanPostProcessor {
+
+  private final ObjectProvider<OpenTelemetry> openTelemetryProvider;
+  private final ObjectProvider<InstrumentationConfig> configProvider;
+
+  public RestClientBeanPostProcessorSpring4(
+      ObjectProvider<OpenTelemetry> openTelemetryProvider,
+      ObjectProvider<InstrumentationConfig> configProvider) {
+    this.openTelemetryProvider = openTelemetryProvider;
+    this.configProvider = configProvider;
+  }
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) {
+    if (bean instanceof RestClient restClient) {
+      return addRestClientInterceptorIfNotPresent(
+          restClient, openTelemetryProvider.getObject(), configProvider.getObject());
+    }
+    return bean;
+  }
+
+  private static RestClient addRestClientInterceptorIfNotPresent(
+      RestClient restClient, OpenTelemetry openTelemetry, InstrumentationConfig config) {
+    ClientHttpRequestInterceptor instrumentationInterceptor = getInterceptor(openTelemetry, config);
+
+    return restClient
+        .mutate()
+        .requestInterceptors(
+            interceptors -> {
+              if (interceptors.stream()
```

Comment:
perhaps we should use something like in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/15546 to avoid rebuilding the client when interceptor wasn't added?